x-pack/plugin/stack/src/javaRestTest/java/org/elasticsearch/xpack/stack/EcsDynamicTemplatesIT.java

```diff
@@ -189,6 +189,14 @@ public void testFlattenedFieldsWithoutSubobjects() throws IOException {
         verifyEcsMappings(indexName);
     }
 
+    public void testWithDateDetectionDisabled() throws IOException {
```

Comment:
Can we test with date detection disabled by default, or use `randomBoolean()` if we want to test with both enabled and disabled? That way, we also test other cases with date detection. For example, I'm not sure if `testFlattenedFieldsWithinAttributes()` will succeed with date detection set to false because the `path_match` for `threat.enrichments.matched.occurred` doesn't contain a wildcard at the beginning, so it might not apply when it is nested under `attributes`.

Also, wdyt about enhancing `generateTestValue` to also return dates in other formats that should work by default, for example randomly returning either an ISO8601 date, or mills since epoch.

Reply:
Great points!
I'll reiterate on this later or tomorrow

Reply:
Applied your proposals as well as the field match pattern fix.
I don't mind random values but I dread random failures, so I used fixed settings and formats

Reply:
> I don't mind random values but I dread random failures, so I used fixed settings and formats

ES uses randomized testing though the test suite. The background is that you can test a larger permutation of scenarios with that. While it's possible for tests to fail randomly, in this case, that's a feature, not a bug, as that would indicate a real issue that wasn't discovered before. You can deterministically re-run a failed test, providing the seed to get the exact same outcome of "random" decisions. See also https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#only-running-failed-tests.

Reply:
Not a bug for sure, only overhead on analysis of tests failures.
The only value I can see in such randomness is if it reduces test execution costs, otherwise it's always better to test all permutations always. Since I added a minimal test for only the relevant fields and since very few formats can be tested (because only few are supported by default), then the overhead is negligible